src/test/java/ibm/jceplus/junit/base/BaseTestAESGCM.java

```diff
@@ -90,20 +90,26 @@ public class BaseTestAESGCM extends BaseTestJunit5 {
 
     protected KeyGenerator aesKeyGen;
     protected SecretKey key;
-    protected AlgorithmParameters params = null;
+    protected AlgorithmParameters params = null; 
     protected boolean success = true;
     protected Method methodCipherUpdateAAD = null;
     protected Constructor<?> ctorGCMParameterSpec = null;
     protected Method methodGCMParameterSpecSetAAD = null;
-    protected int specifiedKeySize = 0;
     byte[] ivBytes = "123456".getBytes();
     byte[] aadBytes = new byte[16];
 
     @BeforeEach
     public void setUp() throws Exception {
         aesKeyGen = KeyGenerator.getInstance("AES", getProviderName());
-        if (specifiedKeySize > 0) {
-            aesKeyGen.init(specifiedKeySize);
+        int keySize = -1;
+        try {
+            keySize = getKeySize();
+        } catch (RuntimeException e) {
+            // Ignore exception since keysize was not set.
+            // This is OK since we intend to test the default keysize.
+        }
+        if (keySize > 0) {
```

Comment:
This only makes sense if the key is actually set, so should we put it in the try block after getting the key? Just a suggestion. I'm good either way.

Reply:
In this location the value of `keySize` is set to -1 on line 104. We then attempt to set the value and ignore an exception if its thrown since the `getKeySize()` method will throw an exception if it does not have a positive value.

By the time we get to line 111 the value should be `-1` if not set, or a positive value if set by the test. 

Hopefully this make sense?

Reply:
I understand how this works now. I was just saying that checking only makes sense if a key size is obtained, thus why I suggested putting in it into the try block and avoiding it altogether if an exception occurs. But again, more of a stylistic comment here.

Reply:
I was thinking something like: 

```
try {
    int keySize = getKeySize();
    if (keySize > 0) {
        aesKeyGen.init(keySize);
    }
} catch (RuntimeException e) {
    // Ignore exception since keysize was not set.
    // This is OK since we intend to test the default keysize.
}
```